### DISCUSSION TITLE
xbox controller connection issues

### DISCUSSION THREAD

**Discussion by josephadampope** [pybricks/support#1929] [Q&A]
I'm having an issue connection xbox controller 1914 to technic hub. When I launch the program the xbox controller shuts off immediately. The controller is updated.

```python
from pybricks.hubs import TechnicHub
from pybricks.iodevices import XboxController
from pybricks.parameters import Axis, Direction, Port
from pybricks.pupdevices import Motor

# Set up all devices.
py1 = TechnicHub(top_side=Axis.Z, front_side=Axis.X, broadcast_channel=45)
motor = Motor(Port.A, Direction.CLOCKWISE)
controller = XboxController()


# The main program starts here.
while controller.triggers()[1] != 0:
    motor.dc(controller.triggers()[0])

**Comment by BertLindeman** [comment]
Hello,

For readability would you please update your post to add before the python source
 \```python

and after the source: 
\```

These are "Back-ticks" and make the source better visable. So we can better understand it.

Bert

**Reply by josephadampope** [reply]
got it thanks

**Reply by BertLindeman** [reply]
See the difference 😄 
Looking at the program as I see it now, the condition in the while is not True, so the program ends.

You might move this while into a 
```python
while True:
```
with a wait(100) to give the program some slack.

**Reply by BertLindeman** [reply]
Could this be a typo?
```python
while controller.triggers()[1] != 0:
    motor.dc(controller.triggers()[0])
```
Ask if trigger[1] is not zero and then using trigger[0]?

[EDIT] It is hard to debug a program with an xbox controller on a technichub. I cheated by running on a spikehub 😁

**Reply by josephadampope** [reply]
Wait 100 with While true worked perfectly. Thank you.
